client/a8c-for-agencies/components/layout/index.tsx

```diff
@@ -34,20 +39,32 @@ export default function Layout( {
 		: 'a4a-layout__container';
 
 	return (
-		<GuidedTourContextProvider>
-			<Main
-				className={ classNames( 'a4a-layout', className, {
-					'is-with-border': withBorder,
-					'is-compact': compact,
-				} ) }
-				fullWidthLayout={ wide }
-				wideLayout={ ! wide } // When we set to full width, we want to set this to false.
-			>
-				<DocumentHead title={ title } />
-				{ sidebarNavigation }
+		<Main
+			className={ classNames( 'a4a-layout', className, {
+				'is-with-border': withBorder,
+				'is-compact': compact,
+			} ) }
+			fullWidthLayout={ wide }
+			wideLayout={ ! wide } // When we set to full width, we want to set this to false.
+		>
+			<DocumentHead title={ title } />
+			{ sidebarNavigation }
 
-				<div className={ layoutContainerClassname }>{ children }</div>
-			</Main>
+			<div className={ layoutContainerClassname }>{ children }</div>
+		</Main>
+	);
+}
+
+export default function Layout( props: Props ) {
+	const guidedTours = useGuidedTours();
+
+	return (
+		<GuidedTourContextProvider
+			guidedTours={ guidedTours }
+			preferenceNames={ A4A_ONBOARDING_TOURS_PREFERENCE_NAME }
+			eventNames={ A4A_ONBOARDING_TOURS_EVENT_NAMES }
+		>
+			<MainLayout { ...props } />
 		</GuidedTourContextProvider>
```

Comment:
What about the idea of introducing a new property for the Layout component to determine if it has the `GuidedTourContextProvider` or not? A boolean property like `withGuidedTour`. Instead of separating it into "two" (one of them a HOC):

```
return withGuidedTour ? (
    <GuidedTourContextProvider
      guidedTours={ guidedTours }
      preferenceNames={ A4A_ONBOARDING_TOURS_PREFERENCE_NAME }
      eventNames={ A4A_ONBOARDING_TOURS_EVENT_NAMES }
    >
      <MainLayout { ...props } />
    </GuidedTourContextProvider>
  ) : (
    <MainLayout {...props} />
  );
```

Reply:
I'd prefer to keep it simple and avoid introducing new props. Also, we can avoid the call of the `useGuidedTours` hook by separating them. But I'm okay with both 🙂 WDYT?

Reply:
We don't have too many properties... yet :D

![image](https://github.com/Automattic/wp-calypso/assets/9832440/3d40617b-341e-4729-9e19-fa135ec6dd60)

The idea is that, as a developer, I have to first realize that we have two Layout components, and then check the difference between both and why one is MainLayout and the other just Layout without any clue to intuit what each one's purpose is.

> Also, we can avoid the call of the useGuidedTours

Yes sure, the `useGuidedTours` would be inside of the condition. That was an example of the idea.

Reply:
> Yes sure, the useGuidedTours would be inside of the condition. That was an example of the idea.

It cannot be inside the condition even if the props may not be changed. See https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level.

Reply:
true...

Reply:
...mmm this could work, wrapping/encapsulating the hook:

```
function useConditionalGuidedTours( withGuidedTour ) {
  return withGuidedTour ? useGuidedTours() : null;
}
```

```
const guidedTours = useConditionalGuidedTours(withGuidedTour);

  return withGuidedTour && guidedTours ? (
```

Reply:
Introduced the `disableGuidedTour` property by https://github.com/Automattic/wp-calypso/pull/91100/commits/c07b0bb4b3ddfb9cd3a3c45e7e815a996350bc1c

Reply:
mmmm... all right.. and now? :D

```
function LayoutWithGuidedTour( { ...props }: Props ) {
	const guidedTours = useGuidedTours();

	return (
		<GuidedTourContextProvider
			guidedTours={ guidedTours }
			preferenceNames={ A4A_ONBOARDING_TOURS_PREFERENCE_NAME }
			eventNames={ A4A_ONBOARDING_TOURS_EVENT_NAMES }
		>
			<MainLayout { ...props } />
		</GuidedTourContextProvider>
	);
}

export default function Layout( {
	disableGuidedTour = false,
	...props
}: Props & { disableGuidedTour?: boolean } ) {
	if ( disableGuidedTour ) {
		return <MainLayout { ...props } />;
	}

	return <LayoutWithGuidedTour { ...props } />;
}
```

Reply:
Done by https://github.com/Automattic/wp-calypso/pull/91100/commits/4fe0db1ed4b83fa1ada349711c4ca55cfc7f8847

Reply:
Perfect! Now, it's even clearer. 👌